src/main/java/com/atlassian/bitbucket/jenkins/internal/model/BitbucketBuildStatus.java

```diff
@@ -2,39 +2,62 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import javax.annotation.Nullable;
 
 import static java.util.Objects.requireNonNull;
 
 @JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
 public class BitbucketBuildStatus {
 
     private final String description;
+    private final Long duration;
     private final String key;
     private final String name;
+    private final String resultKey;
+    private final String serverIdentifier;
     private final BuildState state;
+    private final TestResults testResults;
     private final String url;
 
     @JsonCreator
-    public BitbucketBuildStatus(@Nullable @JsonProperty("description") String description,
+    public BitbucketBuildStatus(@JsonProperty("description") String description,
+                                @JsonProperty("duration") Long duration,
                                 @JsonProperty("key") String key,
-                                @Nullable @JsonProperty("name") String name,
+                                @JsonProperty("name") String name,
+                                @JsonProperty("resultKey") String resultKey,
+                                @JsonProperty("serverIdentifier") String serverIdentifier,
                                 @JsonProperty("state") BuildState state,
+                                @JsonProperty("testResults") TestResults testResults,
                                 @JsonProperty("url") String url) {
+        requireNonNull(key, "key");
+        requireNonNull(state, "state");
+        requireNonNull(url, "url");
         this.description = description;
-        this.key = requireNonNull(key, "key");
+        this.duration = duration;
+        this.key = key;
         this.name = name;
-        this.state = requireNonNull(state, "state");
-        this.url = requireNonNull(url, "url");
+        this.resultKey = resultKey;
```

Comment:
i think we'll have `ref` on here too

Reply:
Different piece of work, you're not wrong, but not part of this work :D